### PR TITLE
Fix: Prevent NPE with autoUpload=false and multiple files are being uploaded

### PR DESCRIPTION
**Pull request by tulioag**
Fixes: vaadin/vaadin-upload#358

### CODE REVIEW

File: vaadin-upload-flow-parent/vaadin-upload-flow/src/main/java/com/vaadin/flow/component/upload/Upload.java [hunk 1]
```diff
@@ -88,8 +88,12 @@ public Upload() {
         DomEventListener allFinishedListener = e -> {
             JsonArray files = e.getEventData().getArray(elementFiles);
 
-            boolean isUploading = IntStream.range(0, files.length()).anyMatch(
-                    index -> files.getObject(index).getBoolean("uploading"));
+            boolean isUploading = IntStream.range(0, files.length())
+                    .anyMatch(index -> {
+                        final String KEY = "uploading";
+                        JsonObject object = files.getObject(index);
+                        return object.hasKey(KEY) && object.getBoolean(KEY);
```
Comment (sissbruecker) [inline vaadin-upload-flow-parent/vaadin-upload-flow/src/main/java/com/vaadin/flow/component/upload/Upload.java:95]: The construct here is a bit unfortunate, but it seems that there is no way to ask Flow to return an optional or a default value? If we need to do this more often, we might want to introduce a helper for that.

File: vaadin-upload-flow-parent/vaadin-upload-flow-integration-tests/src/test/java/com/vaadin/flow/component/upload/tests/NonImmediateUploadIT.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        UploadElement upload = $(UploadElement.class).waitForFirst();
+        WebElement input = getInput(upload);
+        fillPathToUploadInput(input, file1.getPath(), file2.getPath());
+        WebElement button = getButton(upload, buttonType);
+        button.click();
+        TestBenchElement element = $("span").id("error-handler-message");
+        Assert.assertEquals("No errors", element.getText());
+    }
+
+    private WebElement getButton(UploadElement upload, String buttonType) {
```
Comment (sissbruecker) [inline vaadin-upload-flow-parent/vaadin-upload-flow-integration-tests/src/test/java/com/vaadin/flow/component/upload/tests/NonImmediateUploadIT.java:42]: Had to see this visually to understand that it queries a button from a nested `vaadin-upload-file` element. Maybe the method name should reflect that it queries a button within an upload-file element, e.g. `getUploadFileElementButton`.

**Review by sissbruecker — APPROVED** [review]
Looks good, just added some random thoughts.

File: vaadin-upload-flow-parent/vaadin-upload-flow-integration-tests/src/test/java/com/vaadin/flow/component/upload/tests/NonImmediateUploadIT.java (hunk 2 above)
Comment (tulioag) [reply vaadin-upload-flow-parent/vaadin-upload-flow-integration-tests/src/test/java/com/vaadin/flow/component/upload/tests/NonImmediateUploadIT.java]: Method was renamed

File: vaadin-upload-flow-parent/vaadin-upload-flow/src/main/java/com/vaadin/flow/component/upload/Upload.java (hunk 1 above)
Comment (tulioag) [reply vaadin-upload-flow-parent/vaadin-upload-flow/src/main/java/com/vaadin/flow/component/upload/Upload.java:95]: It is a valid concern, but `JsonObject` is not part of Flow and it does not provide an API using Optional.

**Pull request merged**